### ISSUE TITLE
Streaming responses in diff editor (w/ linter) needlessly inefficient

### ISSUE THREAD

**Issue by cosmix** [cline/cline#557]
### Which API Provider are you using?

Anthropic

### Which Model are you using?

Claude 3.5 Sonnet

### What happened?

It would be much more efficient if the content was streamed without VS.Code constantly trying to diff and lint it.

### Steps to reproduce

1. have cline read and write an existing file
2. see the model changes being streamed (and VS.Code struggling to lint/diff it)

**Comment by pietz** [comment]
No thanks. Get a faster linter. This is obviously not a Cline problem.

**Comment by cosmix** [comment]
@pietz thanks for your thoughtful comment and suggestion. I beg to disagree; linting, syntax highlighting, and diffing code that's being streamed is not adding any value to cline or its users. It's 100% a cline issue. If you enjoy seeing your CPU being used for no reason at all you've got many other options.

**Comment by pietz** [comment]
I use ruff in python, which is written in Rust and I just have zero of the issues you mention. Cline only works with files with less than 300 lines of code well, so it can't take up that much compute.

I like linting while generating. I can see immediately if it uses a function that doesn't exist, which is the number one problem I encounter.

I believe I'm entitled to my opinion here and I would like this feature to stay.

**Comment by cosmix** [comment]
Of course. We're all entitled to our opinion, that's my point—as a response to your 'get a faster linter' suggestion. 

First of all: What you're stating you care about is not linting, it's diffing.

More generally: You may have misunderstood the point: I wasn't suggesting disabling linting, syntax highlighting or diffing when viewing the generated output and comparing it with the existing version of the code. That's absolutely necessary. I'm only suggesting disabling it during the streaming phase, i.e. when the content is constantly changing while being streamed. Once it's done, you should be able to get everything enabled.

**Comment by tsdevau** [comment]
I personally dislike the streaming _"feature"_.  I totally agree with @cosmix on the issue of the editor struggling and bouncing around through the streaming phase.  I much prefer the original approach that just displayed the diff when it was complete.

Despite the streaming being far too erratic to provide any value what-so-ever, regrettably, it's probably a _"feature"_ that's here to stay.  Best we could hope for is a configuration option to disable it.  However, at this stage of Cline's development there seems to be a reluctance towards adding option settings.

**Comment by blistovmhz** [comment]
Yea, a third or fourth here.
I can't even figure out what "streaming" does. I'm not sure if it's broken, or it's just not making sense to me. I'm coming from WS, trying to switch to Cline as WS clearly is having some major issues that seem token throttling related.
In Cline, throwing it at a file with 2000 lines, a fairly complex chunk of code, but one that WS managed to refactor prefectly on the first try on day three after launch, Cline with Sonnet 3.5 also manages to get the correct answer on the first try, but absolutely can't implmement it, despite burning $10 on tokens trying to suss out what's actually going wrong. It can tell me the precise line numbers where it's going to put the modified code (in a single try). It can show me that code, but when it comes to implementation in the editor, it completely cacks out, scrolling slowly through the entire file, 1 line at a time, removing every single line that is not related to our updated code, then it adds the first bit of the code it was going to add, then just stops. No option to accept or reject, and it's LOOKING (10 minutes later) like it's trying to remove 1800 lines of the 2000 lines of code.
Reading through other peoples experiences as of a week or two ago, this didn't seem to be the case, but since the streaming feature was added, there are quite a few people complaining it just doesn't work at all, or is enormously inefficient. I definitely agree with the latter. I don't need to see it scanning each line one by one, highjacking my entire IDE for 10 minutes, to fail for $0.50 in tokens each time.

**Comment by celestial-vault** [comment]
Thanks for the suggestion! This has been consolidated into #4358.
